### ISSUE TITLE
6.1.57-1 Does not wake from sleep/suspend

### ISSUE THREAD

**Issue by bAN0NYM0US** [AdityaGarg8/mbp-16.x-ubuntu-kernel#50]
MacBookPro16,2
Ubuntu T2 23.04 LTS (Updated to 23.04.3 LTS)
Downgraded from 6.5.6 to 6.1.57-1 LTS
Touchbar now works, but laptop will not resume after lid was closed.
Starts up with only touch bar illuminated and nothing else.
After a few seconds the MacBook will reboot on it's own and touchbar will no longer work.
Required manual reboot from within Ubuntu to restore touchbar function.

MacBook Pro is working normally if it is turned off every time before closing the lid.

Clean install, not dual booting, only performed apt updates and kernel downgraded to the LTS kernel.

[UPDATE]
macOS 14.1 Monterey is the last version of macOS this was running.
Just restored back to only macOS, all latest updates done.
Created partition for Ubuntu, ran firmware.sh on macOS
Booted to Ubuntu T2 installer, Installed Ubuntu.
In Ubuntu T2, ran firmware.sh to copy the goods.
Backed up /usr/firmware/brcm
Wiped computer.
Clean install of Ubuntu T2 without macOS
Migrated backed up brcm folder to /usr/firmware/brcm
Overwriting conflicting files
Downgraded to 6.1.59 LTS T2 kernel
Same issue.

MacBookPro16,2 updated with macOS 14.1 (23B73)
Ubuntu T2 22.04.3
6.1.59 LTS T2

**Comment by AdityaGarg8** [comment]
Are you running macOS Sonoma?

**Comment by hadongz** [comment]
I have the same issue, dual boot from macOS Ventura. cannot resume after choosing the suspend menu or closing the lid. tried some fixes from [here](https://askubuntu.com/questions/1404856/not-waking-up-from-suspend-ubuntu-22-04) but no avail

**Comment by AdityaGarg8** [comment]
Is this the latest macOS Ventura, released after Sonoma was released?

**Comment by hadongz** [comment]
> Is this the latest macOS Ventura, released after Sonoma was released?

It's Ventura 13.3 (22E252) released on March 27, 2023, so I think it's not after Sonoma was released.

Is there any workaround? thanks

**Comment by AdityaGarg8** [comment]
What is the firmware version in Controller section, in the system information app on macOS

**Comment by hadongz** [comment]
> What is the firmware version in Controller section, in the system information app on macOS

It's 21P365

**Comment by AdityaGarg8** [comment]
It's the buggy firmware

**Comment by AdityaGarg8** [comment]
The same is seen in Sonoma

**Comment by asavageiv** [comment]
I'm seeing this on Ventura 13.1 with controller firmware 20P2059. Are there any known workarounds?

**Comment by AdityaGarg8** [comment]
For you does it freeze and and restart when you try to resume @asavageiv ?

**Comment by asavageiv** [comment]
Actually, no. This time I booted then selected "sleep" from the menu and it went to a black screen and hard-locked. I had to long-press the laptop to turn it off and reboot.

**Comment by AdityaGarg8** [comment]
What’s your Mac model?

**Comment by asavageiv** [comment]
I'm on MacBookPro16,1 too

**Comment by AdityaGarg8** [comment]
Was suspend ever working for you?

**Comment by asavageiv** [comment]
No, but I just installed for the first time on Oct 12. I'm a new user of T2Linux. I've attached a journalctl log for a boot that ended with a failed suspend. Let me know what other logs would be helpful.

[failed-suspend.txt](https://github.com/t2linux/T2-Ubuntu-Kernel/files/13032319/failed-suspend.txt)

**Comment by AdityaGarg8** [comment]
I guess its broken for you too @asavageiv

**Comment by asavageiv** [comment]
Yeah

**Comment by Rom888** [comment]
I have MacBook Pro 16,1 with Hybrid Graphics (2 GPUs).
In my experience, suspend, shutdown and tiny-dfr work fine if the main GPU is intelGPU and doesn't work with amdGPU.

This command can show the main GPU:
glxinfo | grep "OpenGL renderer"

update: after upgrading to the latest kernel wake up from suspend also doesn't work

**Comment by baerenmarke90** [comment]
I've got the same issue (Ubuntu and Debian). Does anybody test the new macOS Sonoma Beta 14.2? Maybe it fixes something?

**Comment by ThatOneGuy140** [comment]
Have there been any new updates regarding this issue yet?

**Comment by AdityaGarg8** [comment]
Some people have got s2idle sleep working.
